chef_master/source/custom_resources_notes.rst

```diff
@@ -181,9 +181,9 @@ If you do need to write code which mutates the system through pure-Ruby then you
      end
    end
 
-The ``converge_by`` block gets why-run correct and will just touch "/tmp/foo" instead of actually doing it. The ``converge_by`` block is also responsible for setting ``update_by_last_action``.
+When the ``converge_by`` block is run in why-run mode, it will only log ``touch "/tmp/foo"`` and will not run the code inside the block. 
 
-In order to use ``converge_by`` correctly you must ensure that you wrap the ``converge_by`` with an idempotency check otherwise your resource will be updated every time it is used and will always fire notifications on every run.
+The ``converge_by`` block is also responsible for setting ``update_by_last_action``, however it does not deal with idempotency and will set the ``update_by_last_action`` to ``true`` everytime the block runs.  By wrapping this block with an idempotency check like ``converge_if_changed`` or ``unless File.exist?("/tmp/foo")`` this will stop ``update_by_last_action`` from being set.
```

Comment:
This is a run-on sentence, and you've also dropped the fact that it fires notifications on every run when you forget it.   Also `converge_if_changed` wraps a `converge_by` block as well, so those are mutually exclusive, you shouldn't use both.   This should probably read more like:


    The ``converge_by`` block does not do any checking for idempotency and always sets ``updated_by_last_action``.  A ``converge_by`` block that is not wrapped in an idempotency check will always cause the resource to be updated, and will always cause notifications to fire.  A properly written resource should wrap all ``converge_by`` checks with an idempotency check, or the ``converge_if_changed`` API should be used instead.   The ``converge_if_changed`` API wraps a ``converge_by`` block with an idempotency check for you.